qml/pages/CameraUI.qml

```diff
@@ -207,37 +175,227 @@ Page {
+        anchors.centerIn: parent
+
+        GridOverlay {
+            aspect: settings.global.captureMode == "image" ? ratio(camera.imageCapture.resolution) : ratio(camera.videoRecorder.resolution)
+
+            function ratio(resolution) {
+                return resolution.width / resolution.height
             }
         }
-    }
 
-    RoundButton {
-        id: btnCapture
+        Slider {
+            id: zoomSlider
+            anchors.horizontalCenter: parent.horizontalCenter
+            anchors.bottom: parent.bottom
+            width: parent.width * 0.75
+            minimumValue: 1
+            maximumValue: camera.maximumDigitalZoom
+            value: camera.digitalZoom
+            stepSize: zoomStepSize
+            rotation: {
+                // Zoom slider should be slide up to zoom in
+                if (camera._orientation  === 1) return -180
+                else if (camera._orientation === 2) return 0
+                else return controlsRotation }
+
+            onValueChanged: {
+                if (value != camera.digitalZoom) camera.digitalZoom = value;
+            }
+
+            Connections {
+                target: camera
 
-        anchors.verticalCenter: parent.verticalCenter
-        anchors.right: parent.right
-        anchors.rightMargin: Theme.paddingMedium
+                onDigitalZoomChanged: {
+                    zoomSlider.value = camera.digitalZoom;
+                }
+            }
+        }
+
+        Image {
+            id: photoPreview
+            rotation: page.controlsRotation
+            onStatusChanged: {
+                if (photoPreview.status == Image.Ready) {
+                    console.log('photoPreview ready');
+                }
+            }
+        }
+
+        RoundButton {
+            id: btnCapture
+
+            anchors.verticalCenter: parent.verticalCenter
+            anchors.right: parent.right
+            anchors.rightMargin: Theme.paddingMedium
+
+            size: Theme.itemSizeLarge
+            rotation: page.controlsRotation
+
+            image: shutterIcon()
+            icon.anchors.margins: Theme.paddingSmall
+            onClicked: doShutter()
+        }
 
-        size: Theme.itemSizeLarge
-        rotation: page.controlsRotation
+        Rectangle {
+            id: rectFlash
+            anchors.fill: parent
+            opacity: 0
 
-        image: shutterIcon()
-        icon.anchors.margins: Theme.paddingSmall
-        onClicked: doShutter()
+            NumberAnimation on opacity {id:animFlash; from: 1.0; to: 0.0; duration: 200 }
+        }
+
+        Row {
+            anchors.top: parent.top
+            anchors.horizontalCenter: parent.horizontalCenter
+            spacing: Theme.paddingMedium
+            rotation: page.controlsRotation
+            transformOrigin: {
+                if (camera._orientation == 1 || camera._orientation >= 5) return Item.Right
```

Comment:
Maybe these could use some enum for orientation values for better readability, probably Qt would have some suitable already? Same for other places using similar magic values.